test/cpp/end2end/BUILD

```diff
@@ -81,6 +81,26 @@ grpc_cc_test(
     ],
 )
 
+grpc_cc_test(
+    name = "async_end2end_no_xds_test",
+    srcs = ["async_end2end_test.cc"],
```

Comment:
This is a fairly heavyweight test.  Is there a smaller test we can use instead, since all we really care about here is that the build works?

Also, is there any reasonable way to verify that the xDS code is actually not being included here?  I guess if nothing else, we can compare the binary sizes of the test built with xDS and the same test built without it.  But maybe we can even run something to check for xDS symbols in the binary?

Reply:
>This is a fairly heavyweight test. Is there a smaller test we can use instead, since all we really care about here is that the build works?
We would ideally want something like `public_headers_must_be_c89.c` to make sure everything works.

I've been trying to figure out a good way to test this myself. I actually was looking into a way to pass `grpc_no_xds` as a flag from this test instead of using the underlying `:grpc++_no_xds` target directly. Bazel seems to have something called transitions but they do not want to support `--define` for transitions https://docs.bazel.build/versions/master/skylark/config.html#unsupported-native-options

Seems like we should be changing the dependency on --define=grpc_no_xds=true to a custom build setting - https://docs.bazel.build/versions/master/skylark/config.html#users-defined-build-settings

Reply:
Maybe what we want is to add a portability build for this

Reply:
I'm open to any reasonable option, as long as it works both internally and externally.  The build rules are fairly different in both environments, and I had to jump through a bunch of hoops to get it working right when I added support for "--define=grpc_no_xds=true".

Reply:
BTW, I'm also open to saying that there's really no reasonable way to do this right now.  If it's going to take a huge amount of work, then it's probably not worth doing right now.

Reply:
Ack, I've cleaned up this PR to simply fix the targets and separate out the xds specific code. I've manually tested that
`bazel build test/cpp/end2end:end2end_test --define=grpc_no_xds=true` works.

I'm working on getting that included in our testing infra so that it gets checked for all our PR builds.